### PR TITLE
OOP-School-Library: Decorators

### PR DESCRIPTION
**Pull request by Abe1able**
## In this milestone of the project, I have 

- Created an interface called Nameable.
- Made the Person class inherit from Nameable.
- Created a base decorator called Decorator.
- Implemented CapitalizeDecorator and TrimmerDecorator used for capitalizing the given word and trimming the word if its length is greater than 10 respectively.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,5 +1,10 @@
-class Person
+require_relative './nameable'
+require_relative './capitalize_decorator'
+require_relative './trimmer_decorator'
+
+class Person < Nameable
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
```
Comment (rustam-tolipov) [inline person.rb:7]: **You did a great job!**

- [x] However, in the Person class, I cannot find the `correct_name` method. If you want to correct names you need to include them in the Person class. Kindly add it.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @Abe1able,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Highly organized code_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Abe1able) [reply person.rb:7]: Implemented. Thank you for your helpful comment @Rustamxon7

**Review by TadesseDev — APPROVED** [review]
### Hi @Abe1able 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [ ] Good job making the changes requested by the previous code reviewer. ✅
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**